js/ad_manager_vast.js

```diff
@@ -73,6 +74,178 @@ OO.Ads.manager(function(_, $) {
+       * Wrapper responses have been received with no inLine response.
+       */
+      WRAPPER_LIMIT_REACHED:              302,
+
+      /**
+       * No ads VAST response after one or more Wrappers. Also includes
+       * number of empty VAST responses from fallback.
+       */
+      WRAPPER_NO_ADS:                     303,
+
+      /**
+       * General linear error. Video player is unable to display the linear ad.
+       */
+      LINEAR_ADS:                         400,
+
+      /**
+       * TODO: Add support
+       * File not found. Unable to find Linear/MediaFile from URI.
+       */
+      FILE_NOT_FOUND:                     401,
+
+      /**
+       * TODO: Add support
+       * Timeout of MediaFile URI.
+       */
+      MEDIAFILE_TIMEOUT:                  402,
+
+      /**
+       * TODO: Add support
+       * Could not find MediaFile that is supported by this video player, based
+       * on the attributes of the MediaFile element.
+       */
+      MEDIAFILE_UNSUPPORTED:              403,
+
+      /**
+       * TODO: Add support
+       * Problem displaying MediaFile.
+       */
+      MEDIAFILE_DISPLAY_PROBLEM:          405,
+
+      /**
+       * General NonLinearAds error.
+       */
+      NONLINEAR_ADS:                      500,
+
+      /**
+       * TODO: Add support
+       * Unable to display NonLinear Ad because creative dimensions do not
+       * align with creative display area(i.e., creative dimension too large).
+       */
+      NONLINEAR_ADS_DIMENSIONS:           501,
+
+      /**
+       * TODO: Add support
+       * Unable to fetch NonLinearAds/NonLinear resource.
+       */
+      NONLINEAR_ADS_UNABLE_TO_FETCH:      502,
+
+      /**
+       * TODO: Add support
+       * Could not find NonLinear resource with supported type.
+       */
+      NONLINEAR_ADS_RESOURCE_UNSUPPORTED: 503,
+
+      /**
+       * TODO: Add support
+       * General CompanionAds error.
+       */
+      COMPANION_ADS:                      600,
+
+      /**
+       * TODO: Add support
+       * Unable to display companion because creative dimensions do not fit
+       * within Companion display area (i.e., no available space).
+       */
+      COMPANION_ADS_DIMENSIONS:           601,
+
+      /**
+       * TODO: Add support
+       * Unable to display Required Companion.
+       */
+      COMPANION_ADS_UNABLE_TO_DISPLAY:    602,
+
+      /**
+       * TODO: Add support
+       * Unable to fetch CompanionAds/Companion resource.
+       */
+      COMPANION_ADS_UNABLE_TO_FETCH:      603,
+
+      /**
+       * TODO: Add support
+       * Could not find Companion resource with supported type.
+       */
+      COMPANION_ADS_RESOURCE_UNSUPPORTED: 604,
+
+      /**
+       * TODO: Add support
+       * Undefined error.
+       */
+      UNDEFINED:                          900,
+
+      /**
+       * TODO: Add support
+       * General VPAID error.
+       */
+      VPAID:                              901
```

Comment:
I was under the impression that, that was what Francisco was working on. Regardless, Vast 3.0 makes references to VPAID, but does not provide any specifications; it does explicitly say that there is a separate IAB guideline for it. Also, VPAID support is not part of the "Minimal Compliance" section in the VAST 3.0 specs.